src/components/shared/FileUpload/FileUpload.tsx

```diff
@@ -269,10 +268,9 @@ const FileUpload = ({
   );
 };
 
-const enhance = compose<
-  AsFieldArrayEnhancedProps<Props> & ForwardedRefProps,
-  Props
->(
+FileUpload.displayName = displayName;
+
+const enhance = compose(
```

Comment:
I'm wondering if the `asFieldArray` could be refactored into a HOC/hook which would allow us to go away from this `compose` pattern?

Reply:
I'd like that very much :)

Reply:
I think I have tried to modify it but there were some problems but don't recall exactly. I have opened another issue for it not to be a drag on this PR #106